### DISCUSSION TITLE
Can got have adapters ?

### DISCUSSION THREAD

**Discussion by KaKi87** [sindresorhus/got#1965] [Q&A] [answered]
Hello,

I created [`axios-tauri-adapter`](https://www.npmjs.com/package/axios-tauri-adapter) which is an `axios` adapter for Tauri, and I'm wondering if I could do the same thing for `got` ?

Thanks

**Comment by sindresorhus** [comment, marked as answer]
Got has something similar, [hooks](https://github.com/sindresorhus/got/blob/main/documentation/9-hooks.md) and [handlers](https://github.com/sindresorhus/got/blob/main/documentation/10-instances.md). [Examples.](https://github.com/sindresorhus/got/blob/main/documentation/examples/advanced-creation.js)

**Reply by KaKi87** [reply]
I'm sorry, I thought that `got`, like [`axios`](https://github.com/axios/axios), had both Node *and* browser support, which isn't the case.

In fact, my question was related to the fact that `axios` uses [`http`](https://nodejs.org/api/http.html) in Node *and* [`XMLHttpRequest`](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest) in browser, using a modular system called [adapters](https://github.com/axios/axios/tree/master/lib/adapters).

This allowed me to create my own adapter for [Tauri](https://tauri.studio/), which is a Rust-based ultra-lightweight alternative to [Electron](https://www.electronjs.org/), featuring its own [`http`](https://tauri.studio/docs/api/js/modules/http/) module.

The goal is that any module using `axios` that was meant to work only on Node or browser can automagically work in other environments.

So, I'm wondering if I could achieve the same thing with `got`, because there are some modules that uses this HTTP client that I would like to be able to use in non-Node environments.

Thanks

**Reply by szmarczak** [reply]
You're looking for https://github.com/sindresorhus/ky

**Reply by KaKi87** [reply]
You mean it's the exact same API but for browsers ?

**Reply by szmarczak** [reply]
Almost the same, yes.

**Reply by KaKi87** [reply]
Alright, thanks !
